samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs

```diff
@@ -109,64 +154,328 @@ private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvok
+        {
+            // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+            CreatThumbnailCardData cardData = JsonConvert.DeserializeObject<CreatThumbnailCardData>(action.Data.ToString());
+            var card = new ThumbnailCard
+            {
+                Title = "ID: " + cardData.EmpId,
+                Subtitle = "Name: " + cardData.EmpName,
+                Text = "E-Mail: " + cardData.EmpEmail,
+                Images = new List<CardImage> { new CardImage("https://3er1viui9wo30pkxh1v2nh4w-wpengine.netdna-ssl.com/wp-content/uploads/prod/2014/10/MSFT_logo_rgb_C-Gray-768x283.png") },
+            };
+
+            var attachments = new List<MessagingExtensionAttachment>();
+            attachments.Add(new MessagingExtensionAttachment
+            {
+                Content = card,
+                ContentType = ThumbnailCard.ContentType,
+                Preview = card.ToAttachment(),
+            });
+
+            return new MessagingExtensionActionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    AttachmentLayout = "list",
+                    Type = "result",
+                    Attachments = attachments,
+                },
+            };
+        }
+
+        private MessagingExtensionActionResponse SendHTMLResponse(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+            CreatHeroCardData cardData = JsonConvert.DeserializeObject<CreatHeroCardData>(action.Data.ToString());
+            var card = new HeroCard
+            {
+                Title = "User Name: " + cardData.UserName,
+                Text = "Successfully logged into " + cardData.UserName + " profile",
+            };
+
+            var attachments = new List<MessagingExtensionAttachment>();
+            attachments.Add(new MessagingExtensionAttachment
+            {
+                Content = card,
+                ContentType = HeroCard.ContentType,
+                Preview = card.ToAttachment(),
+            });
+
+            return new MessagingExtensionActionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    AttachmentLayout = "list",
+                    Type = "result",
+                    Attachments = attachments,
+                },
+            };
+        }
+
+        private MessagingExtensionActionResponse SendAdaptiveCardResponse(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            var createCardData = ((JObject)action.Data).ToObject<CreateCardData>();
+            AdaptiveCard adaptiveCard = new AdaptiveCard(new AdaptiveSchemaVersion(1, 0));
+            adaptiveCard.Body = new List<AdaptiveElement>()
```

Comment:
Please move this code to a separate class.  TeamsMessagingExtensionsActionBot is too large, and difficult to follow.